### ISSUE TITLE
Curator writes in Stderr instead of Stdout

### ISSUE THREAD

**Issue by dominikmueller** [elastic/curator#121]
Hi,

i'm experiencing problems with curator running as a cronjob.
Cron sends emails as soon as a job writes to stderr.
So far so good - but as curator seems to write its log to stderr when there is no logfile defined, cron sends emails each time the job is running.
I think this might be a bug?

Here are the code parts where stderr is used:
https://github.com/elasticsearch/curator/search?q=stderr&ref=cmdform

kind regards
Dominik

**Comment by mark-casey** [comment]
Is there any particular benefit of having `STDERR` as the output? I can open a new issue if desired, but thread resurrection seemed a less cluttered option in this case.

**Comment by untergeek** [comment]
I will make this change for the next release.

**Comment by mark-casey** [comment]
Aww, and I'd pulled together research and everything!   :)

In case what I found is useful for someone upgrading and getting this change (or for any in the future who are doubtful about it) I still want to mention a few findings...

I tested a cronjob on Ubuntu 15.04 Vivid and RHEL7 that ran the date command and got results counter to what @dominikmueller saw. As expected it just printed the date to stdout, but I still got the cron emails. So it is likely that Curator will continue to email after this change if the output is not being redirected away. Obviously this may not hold true on some older releases or other distros, but it is better than the idea that this change will silence most people's output by default (which would be rather scary indeed).

There is also Chuck Houpt's Cronic (http://habilis.net/cronic/) and my spiritual successor to it "Foilhat" (repo is on this account) which can help you only get mail when there is output to stderr or the return code is non-zero.

Finally, for any still uncertain I wanted to share a trivia (and there are many) that rsync prints usage to stderr when there are no arguments but prints almost identical usage output to stdout when the help flag is used (i.e.: because you asked them for help, the usage output is no longer an error).
